waspc/data/Generator/templates/react-app/src/auth/forms/internal/common/LoginSignupForm.tsx

```diff
@@ -184,41 +198,45 @@ export const LoginSignupForm = ({
         </OrContinueWith>
       {=/ areBothSocialAndPasswordBasedAuthEnabled =}
       {=# isAnyPasswordBasedAuthEnabled =}
-        <Form onSubmit={onSubmit}>
+        <Form onSubmit={hookFormHandleSubmit(onSubmit)}>
           {=# isUsernameAndPasswordAuthEnabled =}
           <FormItemGroup>
             <FormLabel>Username</FormLabel>
             <FormInput
+              {...register('username', {
+                required: 'Username is required',
+              })}
               type="text"
-              required
-              value={usernameFieldVal}
-              onChange={e => setUsernameFieldVal(e.target.value)}
               disabled={isLoading}
             />
+            {errors.username && <FormError>{errors.username.message}</FormError>}
```

Comment:
As a side effect of including `react-hook-form` and client side validation becoming trivial, I've included it for existing fields as well.